aikau/src/main/resources/alfresco/services/UploadService.js

```diff
@@ -388,6 +437,35 @@ define(["dojo/_base/declare",
       },
 
       /**
+       * This updates the [uploadHistory]{@link module:alfresco/services/UploadService#uploadHistory} with the
+       * supplied NodeRef. If the NodeRef is already in the history then it will not be added again. If the history
+       * already contains the [maximum number of entries]{@link module:alfresco/services/UploadService#uploadHistorySize}
+       * then the earliest used NodeRef in the history will be removed and the latest added.
+       *
+       * @instance
+       * @param {string} nodeRef The NodeRef to add to the upload history
+       * @since 1.0.34
+       */
+      updateUploadHistory: function alfresco_services_UploadService__updateUploadHistory(nodeRef) {
+         var alreadyInHistory = array.some(this.uploadHistory, function(d) {
+            return d === nodeRef;
+         });
+         if (!alreadyInHistory)
```

Comment:
Do you want to reorder the array if the nodeRef is already in the history? Otherwise you could end up with the history not accurately showing the latest 3 items as the most recently uploaded item might be the next one removed from the list.